### PR TITLE
bpf: monitor drops in NAT64/NAT46 tail calls

### PR DESCRIPTION
**Pull request by nirmoy**
Fixes: #4216

Signed-off-by: Nirmoy Das <ndas@suse.de>

@manalibhutiyani @joestringer am I right about the direction in send_drop_notify here ?

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4312)

### CODE REVIEW

File: bpf/bpf_lxc.c [hunk 1]
```diff
@@ -1011,12 +1012,14 @@ __section_tail(CILIUM_MAP_CALLS, CILIUM_CALL_NAT46) int tail_ipv4_to_ipv6(struct
 	int ret;
 
 	if (!revalidate_data(skb, &data, &data_end, &ip4))
-		return DROP_INVALID;
+		return  send_drop_notify(skb, SECLABEL, 0, 0, 0, DROP_INVALID, TC_ACT_SHOT,
```
Comment (tgraf) [inline bpf/bpf_lxc.c]: Please introduce an inline function which calls revalidate_data() and IPv4_to_ipv6() and call send_drop_notify() in the error handler when calling that new function. Similar to how tail_handle_ipv6() does it. This way we do not inline the send_drop_notify() code twice in the prog section.

**Review by tgraf — CHANGES_REQUESTED** [review]
Direction for metrics are correct!

File: bpf/bpf_lxc.c (hunk 1 above)
Comment (nirmoy) [reply bpf/bpf_lxc.c]: Thanks for reviewing, I updated with a new inline function.

**Review by tgraf — APPROVED** [review]
Great!

**Comment by tgraf** [comment]
test-me-please

**Review by manalibhutiyani — APPROVED** [review]
looks good

**Pull request merged**